cmd/volume-data-source-validator/main.go

```diff
@@ -46,10 +46,9 @@ import (
 
 // Command line flags
 var (
-	kubeconfig   = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
-	resyncPeriod = flag.Duration("resync-period", 60*time.Second, "Resync interval of the controller.")
```

Comment:
Removing a command line flag is a breaking change that would require a major version bump. Not sure whether that is a concern for volume-data-source-validator?

Reply:
It might be safer to change the default to zero, then if the controller turns out to have bugs that can be fixed with a resync, then that's still possible without updating.

Reply:
My concern with making it configurable is that you definitely get log spam and corrupted metrics if the resync period is non-zero and we don't put the resource version check in the update handler. So we have to put the resource version check back in if we make it configurable, even if it defaults to zero.

Reply:
It's not log spam if the user explicitly enables a non-zero period, is it? We can extend the flag help message to explain the implications better.

Non-zero resync period and then suppressing the effect makes no sense.

Reply:
Okay you've convinced me that if we make it configurable it's safe to not add the resource version check.

I'm still very suspicious that there's ever a situation where you'd want to set the value to non-zero. I'm worried we're supporting an old superstition about how shared informers work. From what I've read, the resync is now specifically only to trigger actions that involve polling an external (unwatchable) resource.

Reply:
@pohly This is the thread I read, and it suggested that at one time relist was necessary to work around some bugs. Later on relist was replaced by resync, and then later still resync became mostly unnecessary.

https://groups.google.com/g/kubernetes-sig-api-machinery/c/PbSCXdLDno0

Reply:
I remember that one - I think I came across it when researching this myself a while back.